enhanced_enums/pubspec.lock

```diff
@@ -0,0 +1,334 @@
+# Generated by pub
```

Comment:
We shouldn't need this, right?

Reply:
The lock file in general, you mean?

Reply:
The docs agree with you:

https://dart.dev/guides/libraries/private-files

As a person who arrived at Dart through Flutter, I'm used to checking in the lock file for pretty much everything, whereas you and Parker (as true Dart coders) are probably used to leaving it out more often than including it.

Looks like you're correct, though, so +1 for yanking it. 😄

Reply:
That guideline is for libraries, and this, unless my reading is incorrect, isn't a library. 

My understanding is we check in `pubspec.lock` files for executable samples as a way of making it really easy for people to get a working version of the code. `dart pub get` will pull a tested set of dependencies that we know work together.

Please revert.

Reply:
Right now, the convention in this repo is *not* to commit the lockfiles.

Reply:
Thanks! I will follow the repo convention then and remove the file.